docs/specs/xref.yml

```diff
@@ -2198,3 +2223,221 @@ outputs:
     {"message_severity":"warning","code":"xref-not-found","message":"Cross reference not found: 'a'.","file":"c.md"}
   c.json: |
     { "conceptual": "<p>Link to <span class=\"no-loc\" dir=\"ltr\" lang=\"en-us\"> &lt;xref:a&gt;</span></p>" }
+---
+# [xrefmap][hostNameMapping] output xrefmap with legacy hostname as new hostname may be not available
+# in despite of config: hostname is legacy value
+repos:
+  https://docs.com/test-uid-conceptual#live:
+    - files:
+        docfx.yml: |
+          hostName: docs.microsoft.com
+          basePath: /base_path
+          hostNameMapping: 
+            docs.microsoft.com: new-docs.microsoft.com
+        docs/a.md: |
+          ---
+          title: Title from yaml header a
+          uid: a
+          ---
+        docs/b.md: |
+          ---
+          title: Title from yaml header b
+          uid: b
+          ---
+        docs/c.md: Link to @a
+outputs:
+  base_path/docs/c.json: | 
+    {"conceptual":"<p>Link to <a href=\"a\">Title from yaml header a</a></p>\n"}
+  base_path/docs/a.json:
+  base_path/docs/b.json:
+  .xrefmap.json: | 
+    {
+      "references":[
+        {
+          "uid": "a",
+          "href": "https://docs.microsoft.com/base_path/docs/a",
+          "name": "Title from yaml header a"
+        },
+        {
+          "uid": "b",
+          "href": "https://docs.microsoft.com/base_path/docs/b",
+          "name": "Title from yaml header b"
+        }
+      ]
+    }
+---
+# [xrefmap][hostNameMapping] output xrefmap with legacy hostname as new hostname may be not available
+# in despite of config: hostname is new value
+repos:
+  https://docs.com/test-uid-conceptual#live:
+    - files:
+        docfx.yml: |
+          hostName: new-docs.microsoft.com
+          basePath: /base_path
+          hostNameMapping: 
+            docs.microsoft.com: new-docs.microsoft.com
+        docs/a.md: |
+          ---
+          title: Title from yaml header a
+          uid: a
+          ---
+        docs/b.md: |
+          ---
+          title: Title from yaml header b
+          uid: b
+          ---
+        docs/c.md: Link to @a
+outputs:
+  base_path/docs/c.json: | 
+    {"conceptual":"<p>Link to <a href=\"a\">Title from yaml header a</a></p>\n"}
+  base_path/docs/a.json:
+  base_path/docs/b.json:
+  .xrefmap.json: | 
+    {
+      "references":[
+        {
+          "uid": "a",
+          "href": "https://docs.microsoft.com/base_path/docs/a",
```

Comment:
Similarly we should always output with host name, i.e. `new-docs.microsoft.com` in this case.

Reply:
why is that? I would prefer hostnameMapping as the source of truth, we should always respect it first.